modules/currency_denomination/includes/commerce_pos_currency_denomination.admin.inc

```diff
@@ -0,0 +1,359 @@
+    '#type' => 'textfield',
+    '#size' => 3,
+    '#maxlength' => 3,
+    '#default_value' => NULL,
+    '#title' => t('Denomination Type Currency Code'),
+    '#description' => t('Currency code should be in the following format: "USD".'),
+    '#required' => TRUE,
+  );
+
+  $form['actions']['submit'] = array(
+    '#type' => 'submit',
+    '#value' => t('Save'),
+  );
+
+  return $form;
+}
+
+/**
+ * Validate handler for adding new denomination types.
+ */
+function commerce_pos_currency_add_denomination_type_form_validate($form, &$form_state) {
+  // Check if the denomination type already exists in the db.
+  $result = db_select('commerce_pos_currency_denomination_type', 't')
+    ->fields('t', array('currency_id'))
+    ->condition('currency_code', $form_state['values']['commerce_pos_denomination_type_currency_code'])
+    ->execute()
+    ->fetchField();
+
+  if ($result) {
+    form_set_error('', t('This denomination type already exists in the system.'));
+  }
+}
+
+/**
+ * Submit handler for adding new denomination types.
+ */
+function commerce_pos_currency_add_denomination_type_form_submit($form, &$form_state) {
+  db_insert('commerce_pos_currency_denomination_type')
+    ->fields(array(
+      'currency_code' => $form_state['values']['commerce_pos_denomination_type_currency_code']
+    ))
+    ->execute();
+
+  drupal_set_message(t('The denomination type has been successfully added.'));
+  drupal_goto('admin/commerce/config/pos/denomination');
+}
+
+/**
+ * Delete a denomination type.
+ */
+function commerce_pos_currency_delete_denomination_type_form($form, &$form_state, $currency_id) {
+  // Get the denomination type info.
+  $form_state['denomination_type_info'] = commerce_pos_currency_denomination_get_currency_types_from_id(array($currency_id));
+
+  // If the denomination type is part of the enabled denomination types, then,
+  // don't let the user delete it.
+  $configured_denom_types = variable_get('commerce_pos_default_currency_denomination_types');
+  if (isset($configured_denom_types[$form_state['denomination_type_info'][$currency_id]->currency_code])
+    && $configured_denom_types[$form_state['denomination_type_info'][$currency_id]->currency_code] === $form_state['denomination_type_info'][$currency_id]->currency_code) {
+    form_set_error('', t('This denomination type is currently enabled. Please disable it first before deleting.'));
+    return;
+  }
+
+  if (!empty($form_state['denomination_type_info'])) {
+    $form_state['denomination_type_info'] = $form_state['denomination_type_info'][$currency_id];
+
+    return confirm_form(
+      $form,
+      t('Are you sure you want to delete currency denomination type @type?', array(
+        '@type' => $form_state['denomination_type_info']->currency_code,
+      )),
+      'admin/commerce/config/pos/denomination',
+      t('All denominations under this type will be deleted. This action cannot be undone.'),
+      t('Delete'),
+      t('Cancel')

```

Comment:
Really minor but I've noticed this a few places, there should be a trailing comma for multi-line arrays.